File: InterviewPrep/maxSubArr.js
var maxSubArray = function(nums) {
    let max = nums[0]

    for (let i = 0; i < nums.length; i++) {
        let currSum =0;
        for (let j = i; j < nums.length; j++) {
            currSum += nums[j]
            max = Math.max(max,currSum)
        }
    }

    return max
};

for (let index = 0; index < array.length; index++) {
   for (let j = index; j < array.length; j++) {

    
   }
    
}

console.log(maxSubArray([-1]))

File: InterviewPrep/maxSubArr.py
def maxSubArr(nums):
    sub_max = nums[0]
    curr_max = 0
    for i in nums:
            curr_max += i
            if curr_max < i:
               curr_max = i
            sub_max = max(sub_max,curr_max)
        
    return sub_max